catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -522,6 +531,127 @@ public void init() {
             return "";
         });
 
+        post("/annotations", (req, res) -> {
+            Map<String, Object> incoming = JsonFactory.create()
+                    .parser()
+                    .parseMap(req.body());
+            String workspaceId = incoming.get("workspace")
+                    .toString();
+            String queryId = incoming.get("parent")
+                    .toString();
+            String annotation = incoming.get("note")
+                    .toString();
+            String user = getSubjectEmail();
+            if (user == null) {
+                return util.getResponseWrapper("error",
+                        "You are not authorized to create notes! A user email is required. "
+                                + "Please ensure you are logged in and/or have a valid email registered in the system.");
+            }
+            if (StringUtils.isBlank(annotation)) {
+                return util.getResponseWrapper("error", "No annotation!");
+            }
+            NoteMetacard noteMetacard = new NoteMetacard(queryId, user, annotation);
+
+            Metacard workspaceMetacard = util.findWorkspace(workspaceId);
+
+            if (workspaceMetacard == null) {
+                return util.getResponseWrapper("error", "Cannot find the workspace metacard!");
+            }
+
+            util.copyAttributes(workspaceMetacard, SECURITY_ATTRIBUTES, noteMetacard);
+
+            Metacard note = saveMetacard(noteMetacard);
+
+            SecurityLogger.auditWarn(String.format(
+                    "Attaching an annotation to a resource: resource=%s annotation=%s",
+                    workspaceId,
+                    noteMetacard.getId()), SecurityUtils.getSubject());
+
+            Map<String, String> responseNote = NoteUtil.getResponseNote(note);
+            if (responseNote == null) {
+                throw new IllegalStateException("An error occurred while storing the annotation");
```

Comment:
❗️ We should have an `exception(IllegalStateException.class, (ex, req, res) -> {...});` block to handle these exceptions

Reply:
I converted all the IllegalStateExceptions to `return util.getResponseWrapper("error", ...)`